cocos2d/core/platform/CCTypes.js

```diff
@@ -354,12 +371,29 @@ cc.FontDefinition = function () {
     _t.strokeEnabled = false;
     _t.strokeStyle = cc.color(255, 255, 255, 255);
     _t.lineWidth = 1;
+    _t.lineHeight = _t.fontSize;
+    _t.fontStyle = "normal";
+    _t.fontWeight = "normal";
 
     _t.shadowEnabled = false;
     _t.shadowOffsetX = 0;
     _t.shadowOffsetY = 0;
     _t.shadowBlur = 0;
     _t.shadowOpacity = 1.0;
+
+    //properties mapping:
+    if(properties && properties instanceof Object){
+         for(var key in properties){
+             _t[key] = properties[key];
+         }
+    }
+};
+/**
+ *
+ * Web ONLY
+ * */
+cc.FontDefinition.prototype.getCanvasFontStr = function(){
```

Comment:
getCanvasFontStr() is a Member function of cc.FontDefinition used by cc.LabelTTF & cc.LabelTTF.RenderCmd, shouldnt it remain public? Unless you want to make sure it is not visible by user then i understand.